### ISSUE TITLE
installing on debian12

### ISSUE THREAD

**Issue by schloegl** [formicidae-tracker/myrmidon#257]
**Is your feature request related to a problem? Please describe.**

I'm trying to install myrmidon on Debian12, but my attempts to install from source or with miniforge failed so far. It seems that myrmidon requires a tbb that is much older than what debian or miniforge are supporting,.

**Describe the solution you'd like**

I'd like to install myrmidon with a free software stack (anaconda/miniconda is not free and therefore not an option)

**Describe alternatives you've considered**

I've tried to compile from source, using libopencv, libasia, libprotobuf from debian, this fails with
    tbb/pipeline.h 
not available. changing to tbb/2020_U2 fails, because libopencv would need to be recompiled,

trying miniforge3, it fails with 
conda install 
fails with  "package not found"  tbb=20202  error.

**Additional context**

**Comment by atuleu** [comment]
Could you try to the dev/v0.9.0 branch.

Dependencies on OpenCV have been completely removed on that branch, as it does indeed not play nicely with tbb

Tested on Ubuntu / Pop OS 22.04. Debian 12 should work, as I think ubuntu just upstream the debian package from unstable without modifications.

Conda/Snap may be discontinued in the next month, but it kind of depends on other tasks. If you have the technical know-how, I would happily take any help to port the conda package to a simple python wheel with on manylinux and drop support for R bindings (R is not the right tool for large datasets). From dev/v0.9.0 it should be easier, as big dependencies have been removed (mainly OpenCV).

**Comment by schloegl** [comment]
Thanks. I've been able to compile and install the library from dev/v0.9.0 branch using with 
`cmake -DCMAKE_INSTALL_PREFIX=${PREFIX}  .. && make -j && make install `
this produced $PREFIX/{include,lib} , but no binary (I was expecting bin/fort-studio)

(remark: $PREFIX is /usr/local by default, but I use a different target directory)

Next, I tried to compile the python bindings with the following steps: 

1.  setting up a venv based on python/3.11, and installing  
`pip install setuptools wheel ninja setuptools-git-versioning numpy
`
2. cd myrmidon/bindings/python$ pip install . 
with CPATH, LD_LIBRARY_PATH,CMAKE_PREFIX_PATH,PKG_CONFIG_PATH set to the corresponding directories in my $PREFIX
This failed and I had to copy 
cp -r ../../build/_deps/fort-charis-src/src/fort/time $PREFIX/include/fort/
cp -r ../../build/_deps/fort-charis-src/src/fort/video $PREFIX/include/fort/
cp  ../../build/_deps/fort-charis-build/src/fort/time/libfort-time.so* $PREFIX/lib/
cp  ../../build/_deps/fort-charis-build/src/fort/video/libfort-video.so* $PREFIX/lib/
(the "make install" above seems to miss these)

3) Now, the `pip install .' is working, and 
  python -c "import py_fort_myrmidon"
runs w/o error. 

So, this seems to work. What I do not see is a GUI as in "GUI to analyse ant tracking data ". 
What do I miss ?

**Comment by atuleu** [comment]
Look at the cmake compile option (using ccmake or another tool). There are
options to select studio and the python bindings). By default those are off.

Le ven. 25 oct. 2024 à 22:20, Alois Schlögl ***@***.***> a
écrit :

**Comment by schloegl** [comment]
I see. I used ccmake and enabled python-binding, r-bindung and fort-studio, etc, and had to apply the attached patch. 
[0001-myrmidon-add-qpainterpath.patch.gz](https://github.com/user-attachments/files/17530510/0001-myrmidon-add-qpainterpath.patch.gz)


Still, I run into this error:
```
/myrmidon/src/analyze-video/main.cpp:6:10: fatal error: opencv2/imgcodecs.hpp: No such file or directory
    6 | #include <opencv2/imgcodecs.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.

```
/usr/include/opencv4/opencv2/imgcodecs.hpp is avaible through "apt install libopencv-imgcodecs-dev", but it seems this is not picked up. 

Any idea how this can be addressed ?

**Comment by schloegl** [comment]
It seems this was missing.  
   -I/usr/include/opencv4/ -L/usr/lib/x86_64-linux-gnu/ -lopencv_core -lopencv_videoio -lopencv_highgui

I got this from here:
   https://stackoverflow.com/questions/71799459/undefined-reference-to-cvmatmat
   
When adding this, I was able to compile and install fort-studio. However, I do not know how to make cmake aware of this.

**Comment by atuleu** [comment]
In the `dev/v0.9.0` there is no dependency on OpenCV (at least in the C/C++ sense). However the python tests depends on `python3-opencv`. 

I made a few changes to support different TBB 2020 / 2021 api. Please pull from the `dev/v0.9.0` branch and check the `docs/installation.md` instructions. You may need to change `libtbb2-dev`  to `libtbb-dev` on Debian 12 / Ubuntu 24.04.

I tested this on both Ubuntu 24.04 and Debian 12 ( from a pristine installation).

For the python bindings, I may in the comming weeks provide a wheel package ( and no more conda dependencies )

For `fort-studio` I may also provide a flatpack, as snap are not very convenient. Flatpack are available on debian.

**Comment by schloegl** [comment]
enabling BUILD_R causes the errror above.
I've now enabled BUILD_PYTHON, BUILD_SHARED, BUILD_STUDIO, on top of the default settings (as in CCMAKE), and this compiles w/o error. 

However, when trying to run fort-studio, it fails with 
```
$ fort-studio
fort-studio: error while loading shared libraries: libfort-time.so.1: cannot open shared object file: No such file or directory

```
It seems some libraries an not installed in the proper directories.

**Comment by schloegl** [comment]
Yes, I'm using the 2nd option (using CMAKE_INSTALL_PREFIX and LD_LIBRARY_PATH)
